tests/exceptions.js

```diff
@@ -0,0 +1,104 @@
+const expect = require('chai').expect;
+const {dispatch, ramdap: R} = require('../');
+const errors = require('../lib/errors')['en-us'];
+
+describe('parameter exception handling', function () {
+  const obj = {};
+  const noop = () => {};
+
+  it('getDispatchFn is not a function', function () {
+    const testCase = dispatch(noop, 'not a function');
+
+    return testCase(obj).catch((err) => {
```

Comment:
Any time you want to test that an error is successfully in a mocha test, you have to explicitly throw an error after `testCase(obj)`. Otherwise mocha will noticed the Promise resolved successfully (without even going through your `.catch` handler) and the test will false positive.